api/src/pcapi/domain/show_types.py

```diff
@@ -1,158 +1,176 @@
-            {"code": 1308, "label": "Théâtre d'Objet"},
-            {"code": 1309, "label": "Tragédie"},
-            {"code": -1, "label": "Autre"},
+    ),
+    ShowType(
+        code=1400,
+        label="Pluridisciplinaire",
+        children=[
+            ShowSubType(code=1401, label="Performance"),
+            ShowSubType(code=1402, label="Poésie"),
+            ShowSubType(code=-1, label="Autre"),
         ],
-    },
-    {
-        "code": 1400,
-        "label": "Pluridisciplinaire",
-        "children": [
-            {"code": 1401, "label": "Performance"},
-            {"code": 1402, "label": "Poésie"},
-            {"code": -1, "label": "Autre"},
+    ),
+    ShowType(
+        code=1500,
+        label="Autre (spectacle sur glace, historique, aquatique, …)  ",
+        children=[
+            ShowSubType(code=1501, label="Son et lumière"),
+            ShowSubType(code=1502, label="Spectacle sur glace"),
+            ShowSubType(code=1503, label="Spectacle historique"),
+            ShowSubType(code=1504, label="Spectacle aquatique"),
+            ShowSubType(code=-1, label="Autre"),
         ],
-    },
-    {
-        "code": 1500,
-        "label": "Autre (spectacle sur glace, historique, aquatique, …)  ",
-        "children": [
-            {"code": 1501, "label": "Son et lumière"},
-            {"code": 1502, "label": "Spectacle sur glace"},
-            {"code": 1503, "label": "Spectacle historique"},
-            {"code": 1504, "label": "Spectacle aquatique"},
-            {"code": -1, "label": "Autre"},
+    ),
+    ShowType(
+        code=1510,
+        label="Opéra",
+        children=[
+            ShowSubType(code=1511, label="Opéra série"),
+            ShowSubType(code=1512, label="Grand opéra"),
+            ShowSubType(code=1513, label="Opéra bouffe"),
+            ShowSubType(code=1514, label="Opéra comique"),
+            ShowSubType(code=1515, label="Opéra-ballet"),
+            ShowSubType(code=1516, label="Singspiel"),
+            ShowSubType(code=-1, label="Autre"),
         ],
-    },
-    {
-        "code": 1510,
-        "label": "Opéra",
-        "children": [
-            {"code": 1511, "label": "Opéra série"},
-            {"code": 1512, "label": "Grand opéra"},
-            {"code": 1513, "label": "Opéra bouffe"},
-            {"code": 1514, "label": "Opéra comique"},
-            {"code": 1515, "label": "Opéra-ballet"},
-            {"code": 1516, "label": "Singspiel"},
-            {"code": -1, "label": "Autre"},
+    ),
+    ShowType(
+        code=-1,
+        label="Autre",
+        children=[
+            ShowSubType(code=-1, label="Autre"),
         ],
-    },
-    {
-        "code": -1,
-        "label": "Autre",
-        "children": [{"code": -1, "label": "Autre"}],
-    },
+    ),
 ]
 
-SHOW_TYPES_DICT = {show_type["code"]: show_type["label"] for show_type in show_types}
+SHOW_TYPES_DICT = {vars(show_type)["code"]: vars(show_type)["label"] for show_type in show_types}
```

Comment:
Quitte à utiliser une dataclass, est-ce qu'on ne pourrait accéder directement à ses attributs ?

```suggestion
SHOW_TYPES_DICT = {show_type.code: show_type.label for show_type in show_types}
```

Idem plus bas pour `SHOW_SUB_TYPES_DICT`.